src/main/java/org/opensearch/security/securityconf/SecurityRoles.java

```diff
@@ -97,5 +98,7 @@ Set<String> getAllPermittedIndicesForDashboards(
 
     SecurityRoles filter(Set<String> roles);
 
+    SecurityRoles createSecurityRole(String roleName, Set<String> clusterPerms, Map<String, Set<String>> indexPatternToAllowedActions);
```

Comment:
Not necessarily, this would be useful for API Tokens where we can create an in-memory role from the permissions in a token

Reply:
`createSecurityRole` doesn't align with how these configuration has been used before, its meant to be a representation of the security configuration from the security index, not a way to update/generate/create security objects.

Rather than overload SecurityRole with this responsibility, lets make a new component that is initialized in its own way and separately from our existing configuration.  Its strange to have the logic to create these roles inside of configuration model specific code and for its state to be held inside of the privilege evaluator.